src/Database/test.js

```diff
@@ -102,21 +102,21 @@ describe('Batch writes', () => {
       ['update', m2],
     ])
 
-    expect(collectionObserver).toHaveBeenCalledTimes(4)
+    expect(collectionObserver).toHaveBeenCalledTimes(1)
     expect(collectionObserver).toHaveBeenCalledWith([
-      { record: m1, type: CollectionChangeTypes.updated },
-    ])
-    expect(collectionObserver).toHaveBeenCalledWith([
-      { record: m2, type: CollectionChangeTypes.updated },
+        { record: m3, type: CollectionChangeTypes.created },
+        { record: m1, type: CollectionChangeTypes.updated },
+        { record: m4, type: CollectionChangeTypes.created },
+        { record: m2, type: CollectionChangeTypes.updated },
     ])
```

Comment:
I think it would be good to modify this test, so that the `batch` operations are not all on one collection, but are on a few collections, e.g. `batch(tasks.prepareCreate(...), comment1.prepareUpdate(...), task2.prepareUpdate(...), ....)`. It would be best if the operations were mixed in between together. This way we're testing this batching logic: https://github.com/Nozbe/WatermelonDB/pull/271/files#diff-5f2ce5c6b82cec300f716b06aefd9e1aR75

Reply:
https://github.com/Nozbe/WatermelonDB/pull/271/commits/ed2da4abecd110a56820d9aa30c661fcde8992fc